Modules/CalcDefence.lua

```diff
@@ -1113,68 +1114,102 @@ function calcs.defence(env, actor)
-			end
-			output[damageType.."TotalEHP"] = output[damageType.."MaximumHitTaken"] / (1 - output[DamageType.."NotHitChance"] / 100) / (1 - convertedAvoidance / 100)
-			if minimumEHPMode ~= "NONE" then
-				if output[damageType.."TotalEHP"] < minimumEHP then
-					minimumEHP = output[damageType.."TotalEHP"]
-					minimumEHPMode = DamageType
+		output[damageType.."MaximumHitTaken"] = 2147483648
+		for _, damageConvertedType in ipairs(dmgTypeList) do
+			if actor.damageShiftTable[damageType][damageConvertedType] > 0 then
+				local hitTaken = output[damageConvertedType.."TotalPool"] / (actor.damageShiftTable[damageType][damageConvertedType] / 100) / output[damageConvertedType.."BaseTakenHitMult"]
+				if hitTaken < output[damageType.."MaximumHitTaken"] then
+					output[damageType.."MaximumHitTaken"] = hitTaken
+				end
+				
+				if breakdown then
+					t_insert(breakdown[damageType.."MaximumHitTaken"].rowList, {
+						type = s_format("%d%% as %s", actor.damageShiftTable[damageType][damageConvertedType], damageConvertedType),
+						pool = s_format("x %d", output[damageConvertedType.."TotalPool"]),
+						taken = s_format("/ %.2f", output[damageConvertedType.."BaseTakenHitMult"]),
+						final = s_format("x %.0f", hitTaken),
+					})
 				end
-			elseif breakdown then
-				breakdown[damageType.."TotalEHP"] = {
-				s_format("EHP calculation Mode: %s", DamageType),
-				s_format("Maximum Hit taken: %d", output[damageType.."MaximumHitTaken"]),
-				s_format("%s chance not to be hit: %d%%", DamageType, output[DamageType.."NotHitChance"]),
-				s_format("%s chance to not take damage when hit: %d%%", DamageType, convertedAvoidance),
-				s_format("Total Effective Hit Pool: %d", output[damageType.."TotalEHP"]),
-				}
 			end
 		end
+		--output[damageType.."MaximumHitTaken"] = output[damageType.."TotalPool"] / output[damageType.."TakenHitMult"]
+		--if breakdown then
+		--	breakdown[damageType.."MaximumHitTaken"] = { 
+		--	 s_format("Total Pool: %d", output[damageType.."TotalPool"]),
+		--	 s_format("Damage Taken modifier: %.2f", output[damageType.."TakenHitMult"]),
+		--	 s_format("Maximum hit you can take: %d", output[damageType.."MaximumHitTaken"]),
+		--	}
+		--end
 	end
-	if minimumEHPMode ~= "NONE" then
-		for _, damageType in ipairs(dmgTypeList) do
-			local convertedAvoidance = 0
-			for _, damageConvertedType in ipairs(dmgTypeList) do
-				convertedAvoidance = convertedAvoidance + output[damageConvertedType..minimumEHPMode.."DamageChance"] * actor.damageShiftTable[damageType][damageConvertedType] / 100
+	
+	
+	for _, damageType in ipairs(dmgTypeList) do
+		local DamageTypeConfig = env.configInput.EhpCalcMode or "Average"
+		local DamageType = DamageTypeConfig
+		local minimumcttd = -2147483648
+		local minimumEHPMode = "NONE"
+		if DamageTypeConfig == "Minimum" then
+			DamageTypeConfig = {"Melee", "Projectile", "Spell", "SpellProjectile"}
```

Comment:
Why is `DamageTypeConfig` redefined here, then later redefined back to "Minimum"? Why not create a new local variable?